pkg/operator/ceph/cluster/cluster.go

```diff
@@ -304,7 +305,11 @@ func clusterChanged(oldCluster, newCluster cephv1.ClusterSpec, clusterRef *clust
 
 	// any change in the crd will trigger an orchestration
 	if !reflect.DeepEqual(oldCluster, newCluster) {
-		diff := cmp.Diff(oldCluster, newCluster)
+		// resource quantity fields are non-exportable, so we will compare using the public API comparison method
+		resourceCompareOpt := cmp.Comparer(func(x, y resource.Quantity) bool {
```

Comment:
I'm quite sure it is unnecessary to use a custom comparison function here. Have you tried whether using `AllowUnexported` as suggested by the error message fixes this?

From the `go-cmp` docs:
>  If no custom equality functions are used and no Equal method is defined, equality is determined by recursively comparing the primitive kinds on both values, much like reflect.DeepEqual. Unlike reflect.DeepEqual, unexported fields are not compared by default; they result in panics unless suppressed by using an Ignore option (see cmpopts.IgnoreUnexported) or explicitly compared using the AllowUnexported option.

The surrounding `if` block uses `reflect.DeepEqual`, which means that whatever resources are found here are assumed different by the standards of `reflect.DeepEqual`. And using `AllowUnexported` will give `cmp.Diff` behavior "much like reflect.DeepEqual." I read this to mean that it is effectively the same then as `reflect.DeepEqual`.

-----

If using `AllowUnexported` does not fix the issue, then I think we should create a diff function as part of the core Rook API since we are now establishing a pattern of diffing resources in Kubernetes. The desire to diff k8s resources with resource limits established is not unique to Ceph.

Reply:
I can make the change. I was actually heading down that path to start with but figured to limit the impact.